src/System.Private.Uri/src/System/IriHelper.cs

```diff
@@ -138,13 +138,9 @@ internal static bool CheckIsReserved(char ch, UriComponents component)
         //
         internal static unsafe string EscapeUnescapeIri(char* pInput, int start, int end, UriComponents component)
         {
-            char[] dest = new char[end - start];
+            PooledCharArray dest = new PooledCharArray(end - start);
```

Comment:
Can you just use ValueStringBuilder instead?

Reply:
Switched to ValueStringBuilder, however it required an extension of the type's contract because it was designed for a different scenario. Changes are backward compatible.